### PR TITLE
Adding docs for Cloud Structure Run Events

### PR DESCRIPTION
**Pull request by zachgiordano**
- [X] I have read and agree to the contributing guidelines for [submitting new pull requests](https://github.com/griptape-ai/griptape?tab=readme-ov-file#submitting-pull-requests).

## Describe your changes

Adding docs for Cloud Structure Run Events

## Issue ticket number and link

----
📚 Documentation preview 📚: https://griptape--1187.org.readthedocs.build//1187/

### CODE REVIEW

File: docs/griptape-cloud/structures/run-structure.md [hunk 1]
```diff
@@ -1,21 +1,23 @@
 # Running a Structure
 
-Once your Structure is created and deployed, you can run your Structure one of three ways outlined below. You view the output of any of your runs, no matter how you created them, in the `Runs` tab of your Structure.
+Once your Structure is created and deployed, you can run your Structure one of three ways outlined below. You may view the details of any of your runs, no matter how you created them, in the `Runs` tab of your Structure.
+
+To learn more about Structure Run details and output, look at the [Structure Run Events](./structure-run-events.md) documentations.
```
Comment (collindutter) [inline docs/griptape-cloud/structures/run-structure.md:5]: > documentationS

File: docs/griptape-cloud/structures/run-structure.md [hunk 2]
```diff
@@ -1,21 +1,23 @@
...
 
 ## From the Cloud Console
 
 In the cloud console, click on the name of the Structure you wish to run and then go to the `Test` tab. Here you can specify arguments to pass to your Structure run and any run-specific environment variables you need.
 
 When passing arguments through the cloud console, pass each new argument on a new line. For example if your local code is ran with the inputs `-i input_file.txt` then the arguments you would pass in the cloud would be:
 
-```
+```shell
```
Comment (collindutter) [inline docs/griptape-cloud/structures/run-structure.md:13]: I've seen ```bash used elsewhere, do you know what the difference is?

File: docs/griptape-cloud/structures/structure-run-events.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+### User Events
+
+User Events are any Events emitted by your Structure code to the Events API. The recommended approach for emitting those Events is to make use of the
+[Griptape Cloud Event Listener Driver](../../griptape-framework/drivers/event-listener-drivers.md#griptape-cloud) in the Griptape framework.
+
+For a full example of Structure code that makes use of that driver, refer to the [Managed Structure Template](https://github.com/griptape-ai/managed-structure-template/blob/main/structure.py).
+
+#### Structure Run Output
+
+In order for Griptape Cloud to populate the `output` field for your Structure Run, there needs to be a `USER` Event of type `FinishStructureRunEvent`.
```
Comment (collindutter) [inline docs/griptape-cloud/structures/structure-run-events.md:120]: > there needs to be a `USER` Event of type `FinishStructureRunEvent`.

> you must send an `USER` Event of type `FinishStructureRunEvent`.

File: docs/griptape-cloud/structures/run-structure.md (hunk 2 above)
Comment (vachillo) [reply docs/griptape-cloud/structures/run-structure.md:13]: bash is just a specific type of shell. in this example it makes no difference

File: docs/griptape-cloud/structures/run-structure.md [hunk 4]
```diff
@@ -1,21 +1,23 @@
...
 -i
 input_file.txt
 ```
 
 ## From the API
 
-You can run your Structure via the API using CURL or any other code that can make HTTP requests. You will need a [Griptape Cloud API Key](https://cloud.griptape.ai/configuration/api-keys) and the `Structure Invocation URL` which is located on the `Config` tab of your Structure. 
+You can run your Structure via the API using CURL or any other code that can make HTTP requests. You will need a [Griptape Cloud API Key](https://cloud.griptape.ai/configuration/api-keys) and the `Structure Invocation URL` which is located on the `Config` tab of your Structure.
```
Comment (vachillo) [inline docs/griptape-cloud/structures/run-structure.md:20]: hyper nit that you didnt actually add: should be `cURL`

File: docs/griptape-cloud/structures/structure-run-events.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    },
+    "created_at": "2024-09-11T23:15:54.754921Z",
+    "origin": "SYSTEM",
+    "structure_run": "12345678-e125-4c38-b67f-02b82aa443ce"
+}
+```
+
+### User Events
+
+User Events are any Events emitted by your Structure code to the Events API. The recommended approach for emitting those Events is to make use of the
```
Comment (vachillo) [inline docs/griptape-cloud/structures/structure-run-events.md:113]: we do `SYSTEM` everywhere above, should this be `USER`?

File: docs/griptape-cloud/structures/run-structure.md (hunk 1 above)
Comment (zachgiordano) [reply docs/griptape-cloud/structures/run-structure.md:5]: Many much documentations

File: docs/griptape-cloud/structures/run-structure.md (hunk 2 above)
Comment (zachgiordano) [reply docs/griptape-cloud/structures/run-structure.md:13]: I'm not sure, but it seems to render the same (at least for this snippet), and bash is shorter so I'll switch to that!

**Review by vachillo — APPROVED** [review]

**Pull request merged**